### PR TITLE
[DDW-536] Automate code formatting

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR introduces the [prettier](https://prettier.io) formatting tool and sets up handy automation (using [husky](https://www.npmjs.com/package/husky) and [pretty-quick](https://www.npmjs.com/package/pretty-quick)) for basic maintainance tasks that are often forgotten during development:

- Before each commit: **staged** files are reformatted with prettier.
- Before each push: **All files** are linted, flow checked and formatted (in-place) with prettier.

If one of the checks fails, it's not possible to push and the issues have to be fixed 🔥  

## How to Test
In theory nothing should be broken since this is "just" a complete reformatting … but it would be good to have the full app tested 😬 

## For Developers
It's probably a good idea if you setup your IDE to format any file with prettier after saving. It's widely supported and should be relatively easy to setup!

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by DominikGuzei** [comment]
Thanks for the feedback @daniloprates, we need to tweak it 👍 there is actually some integration with eslint and prettier … have to check again

**Comment by DominikGuzei** [comment]
@nikolaglumac @daniloprates so it turned out that i forgot to add `"` around the files glob and that's why it was only formatting styles 😉 now it should be correct and is also properly integrated with eslint.

**Comment by nikolaglumac** [comment]
@daniloprates may I ask you to merge latest develop into this branch? 🙏

**Comment by daniloprates** [comment]
> @daniloprates may I ask you to merge latest develop into this branch? 🙏

@nikolaglumac @DominikGuzei the merge commit was creating some loop conflicts. I had to skip `husky` to merge the files "unprettied", then run pretty and commit again. Up to date, now.

**Comment by nikolaglumac** [comment]
@daniloprates may I kindly ask you to update this branch with the latest develop and fix conflicts and then run the final review? 🙏

**Comment by nikolaglumac** [comment]
@daniloprates are you happy with this PR now?

File: package.json [hunk 1]
```diff
@@ -99,6 +102,8 @@
     "postcss": "7.0.7",
     "postcss-modules": "1.4.1",
     "postcss-modules-values": "2.0.0",
+    "prettier": "^1.16.4",
```
Comment (nikolaglumac) [inline package.json]: @DominikGuzei please freeze the version here 🙏

**Comment by DominikGuzei** [comment]
Yeah @nikolaglumac that's something i forgot to mention yesterday - i went through all eslint settings we had and cleaned up anything irrelevant and kept the parts that we should actually be careful about … so these warnings are all things that should eventually be improved in our codebase (IMO) but are not "errors" per se. I would like to introduce these warnings as part of the coding guidelines 😉 

If you want to you can also look through them and remove something if you don't agree with certain warnings.

**Comment by nikolaglumac** [comment]
@DominikGuzei I believe we should have zero warnings in order to merge this PR. Let's revisit this PR tomorrow when you start working 🙏

**Comment by DominikGuzei** [comment]
Hmm … i don't agree fully, IMO warnings are exactly good for this very purpose: remembering us to refactor certain things but not necessarily **now** 😉 you can see it like deprecation warnings

**Comment by nikolaglumac** [comment]
I don't like deprecation warnings :(

**Comment by DominikGuzei** [comment]
Then i would suggest setting up a task for cleaning up those warnings 😛

**Comment by nikolaglumac** [comment]
Task created: https://iohk.myjetbrains.com/youtrack/issue/DDW-595

**Comment by nikolaglumac** [comment]
@IuliaDolishniak @DmitriiGaico please test and approve this PR so that we can merge it 🙏

**Pull request merged**